src/edit/selection.js

```diff
@@ -355,8 +355,9 @@ class SelectionToken {
   }
 }
 
-function selectionFromDOM(doc, oldHead) {
-  let sel = window.getSelection()
+function selectionFromDOM(pm, oldHead) {
+  let sel = pm.root.getSelection()
+  const doc = pm.doc
   let {pos: head, inLeaf: headLeaf} = posFromDOM(sel.focusNode, sel.focusOffset)
   if (headLeaf > -1 && sel.isCollapsed) {
```

Comment:
`sel.isCollapsed` was broken in my local testing.

I needed to change this to:

```
const isCollapsed = sel.focusNode === sel.anchorNode && sel.focusOffset === sel.anchorOffset;
if (headLeaf > -1 && isCollapsed) {
```